api/backend/blueflood/blueflood.go

```diff
@@ -218,18 +260,21 @@ var samplerMap map[api.SampleMethod]struct {
 
 // Blueflood keys the resolution param to a java enum, so we have to convert
 // between them.
-func bluefloodResolution(r int64) resolution {
+func (c Config) bluefloodResolution(resolution int64, timestamp int64) string {
```

Comment:
I don't think making something like this generic is going to buy us much. It's only like 20 lines of code and used only in once place.